### PR TITLE
mavproxy: init at 1.8.17

### PR DESCRIPTION
**Pull request by lopsided98**
###### Motivation for this change

MAVProxy is a command line ground control station (GCS) for drones using the standard MAVLink protocol (ie. ArduPilot, PX4). Some GUI GCSs are already included in nixpkgs (QGroundControl, apmplanner2), but MAVProxy includes move development focused functionality that is required for certain use-cases, such as performing software-in-the-loop simulation with ArduPilot.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @wucke13 because you have worked on apmplanner2 and QGroundControl, so might be interested in this package

### CODE REVIEW

File: pkgs/applications/science/robotics/mavproxy/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+buildPythonApplication rec {
+  pname = "MAVProxy";
+  version = "1.8.17";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "193hjilsmbljbgj7v6icy3b4hzm14l0z6v05v7ycx6larij5xj2r";
+  };
+
+  propagatedBuildInputs = [ matplotlib numpy pymavlink pyserial setuptools
```
Comment (jonringer) [inline pkgs/applications/science/robotics/mavproxy/default.nix]: please do a package per-line if it's this long
```suggestion
  propagatedBuildInputs = [ 
    matplotlib 
    numpy 
    pymavlink
     pyserial 
     setuptools
```

File: pkgs/development/python-modules/pymavlink/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0y9rz3piddzdjpp7d5w9xi6lc9v9b4p4375a5hrfiphrmhds85i3";
+  };
+
+  propagatedBuildInputs = [ future lxml ];
+
+  # No tests included in PyPI tarball
+  doCheck = false;
```
Comment (jonringer) [inline pkgs/development/python-modules/pymavlink/default.nix]: please check out from github if the tarball doesn't contain any tests

File: pkgs/development/python-modules/pymavlink/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+buildPythonPackage rec {
+  pname = "pymavlink";
+  version = "2.4.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0y9rz3piddzdjpp7d5w9xi6lc9v9b4p4375a5hrfiphrmhds85i3";
+  };
+
+  propagatedBuildInputs = [ future lxml ];
```
Comment (jonringer) [inline pkgs/development/python-modules/pymavlink/default.nix]: future is a python2 only library
```suggestion
  propagatedBuildInputs = [ lxml ] ++ lib.optionals isPy27 [ future ];
```

File: pkgs/development/python-modules/pymavlink/default.nix (hunk 2 above)
Comment (lopsided98) [reply pkgs/development/python-modules/pymavlink/default.nix]: The PyPI tarball contains generated code for the MAVLink messages. They can be generated at build time, but this requires the message definitions found here: https://github.com/mavlink/mavlink/tree/master/message_definitions/v1.0

The PyPI tarball uses a seemingly random commit (found by manually bisecting to find the commit that matched the tarball): https://github.com/mavlink/mavlink/commit/107cf8823bdbdd68053bbd990e25885c935a48fa

I think we should use the same commit as upstream for compatibility, but I'm not sure it's worth the extra maintenance burden just so we can run the tests.

File: pkgs/development/python-modules/pymavlink/default.nix (hunk 3 above)
Comment (lopsided98) [reply pkgs/development/python-modules/pymavlink/default.nix]: `future` is an unconditional `install_requires`, therefore it is still needed on Python 3 to satisfy dependency resolution.

File: pkgs/development/python-modules/pymavlink/default.nix (hunk 2 above)
Comment (jonringer) [reply pkgs/development/python-modules/pymavlink/default.nix]: ah, didn't realize they did some special seeding when distributing, then you can disregard :)

File: pkgs/development/python-modules/pymavlink/default.nix (hunk 3 above)
Comment (jonringer) [reply pkgs/development/python-modules/pymavlink/default.nix]: this is technically incorrect (it should only be needed for python2), but it's an upstream issue. All good to leave it in propagatedBuildInputs, I think the closure size addition is small anyway.

**Review by jonringer — APPROVED** [review]
diff LGTM
commits LGTM
mavproxy executable shows usagea

```
[7 built, 2 copied (83.3 MiB), 15.1 MiB DL]
https://github.com/NixOS/nixpkgs/pull/75640
4 package were built:
mavproxy python27Packages.pymavlink python37Packages.pymavlink python38Packages.pymavlink
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build mavproxy python27Packages.pymavlink python37Packages.pymavlink python38Packages.pymavlink

**Pull request merged**